dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/AbstractMethodProblem.java

```diff
@@ -17,12 +17,36 @@
 package com.google.cloud.tools.opensource.classpath;
 
 /**
- * The referenced {@code methodSymbol} is not implemented in the {@code targetClass} but the class
- * is declared to implement the method by {@code sourceClass}. Such unimplemented methods manifest
- * as {@link AbstractMethodError}s at runtime.
+ * {@code implementationClass} (the source of an invalid reference) does not implement the abstract
+ * method {@code methodSymbol} declared by {@code interfaceOrAbstractClass} (the target of the
+ * reference). Such unimplemented methods manifest as {@link AbstractMethodError}s at runtime.
```

Comment:
@marquiswang I updated the javadoc to clarify `implementationClass` is the source and `interfaceOrAbstractClass` is the target of the reference.

Reply:
This still doesn't work. I don't see how implementationClass is the "source of an invalid reference." There simply is no invalid reference here. There is a problem but it is something else entirely. It is **not** a reference.

Reply:
It is a reference. Implementation class references an interface via `implements` keyword or an abstract class via `extends` keyword. Implementation-wise, it's a reference embed in the constant pool section of a class file.

Reply:
Yes, but this reference is not invalid. It's a valid, found reference. The problem is that the subtype doesn't implement an abstract method. Sources, targets, and references are not the conventionally understood way to describe this sort of problem.

Reply:
I drew a diagram to explain how I recognize that invalid references describe the abstract method problems in the same manner as missing field problems.

<img width="633" alt="Screen Shot 2020-08-20 at 16 33 02" src="https://user-images.githubusercontent.com/28604/90822730-e0d95280-e302-11ea-9b0b-e6162c5e7bd6.png">

I'll setup.a meeting to understand your viewpoint better.